snap/hooks/configure

```diff
@@ -1,158 +1,90 @@
 #!/bin/bash
 
-display_changes=0
 config_changes=0
-kiosk_display="$SNAP_DATA/ubuntu-core-kiosk.display"
-kiosk_config="$SNAP_DATA/ubuntu-core-kiosk.config"
+display_file="$SNAP_DATA/ubuntu-core-kiosk.display"
+config_file="$SNAP_DATA/ubuntu-core-kiosk.config"
 
-# Ensure we have a config file with the fixed options
-if [ ! -e "${kiosk_config}" ]; then
-cat <<EOT > "${kiosk_config}"
-arw-file=
-console-provider=vt
-EOT
-let config_changes+=1
-fi
+config_entry() {
+  if ! grep -q "^[# ]*${1##?[#\ ]}=" "${config_file}"; then
+    cat <<EOT >> "${config_file}"
```

Comment:
If this is only used for the defaults, couldn't we just `<<EOF` the whole thing in one go?

Reply:
Your premise is false: If the file exists, but comes from a different version (or has been edited) we want to fill in any missing options.

Reply:
Why don't we apply the same "DO NOT EDIT" approach to the file, then?

Reply:
I assumed editing is the normal and easiest way to try things out, or set up a one-off system. But the flow could be:

1. Write the `kiosk-config` value into a temporary file
2. merge with the defaults (same as proposed logic, but using the temporary file)
3. diff with the current file
4. Profit

WDYT?

Reply:
Yeah, I think if we give a `snap set` way, allowing for direct file editing would be bound to cause grief.

I would still be happier if we could just `echo $DEFAULTS $USER_CONFIG > config.file`. Can you please find out if Boost.Options has defined behaviour with duplicate keys? It's customary to respect the last entry, hoping that would be the case here.